numpy/core/src/multiarray/ctors.c

```diff
@@ -4986,7 +4986,7 @@ HPyArray_ArangeObj(HPyContext *ctx, HPy start, HPy stop, HPy step, HPy /* PyArra
         PyObject *new;
         PyObject *py_range = HPy_AsPyObject(ctx, range);
         CAPI_WARN("calling PyArray_Byteswap");
-        new = PyArray_Byteswap(py_range, 1);
+        new = PyArray_Byteswap((PyArrayObject *)py_range, 1);
```

Comment:
Should we perhaps do `PyArrayObject *py_range = (PyArrayObject *) HPy_AsPyObject(ctx, range)` where pyrange is defined?

Reply:
I think this was has fewer casts, since most places this is used prefer a `PyObject*`